### PR TITLE
Add Sweep job to quickstarts

### PR DESCRIPTION
**Pull request by Anubha98**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add Sweep job to quickstarts
* Tested Experiment: https://ml.azure.com/experiments/id/69f780e4-cc20-4886-a6bf-778238d143a7/runs/sweepjob_using4?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/guptaaman/providers/Microsoft.MachineLearningServices/workspaces/anubha-wrspc&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

### CODE REVIEW

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/script/main.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# imports
```
Comment (shivamMg) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/script/main.py:1]: I think the folder should be named `scripts` - as mentioned in [best practices](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep [hunk 2]
```diff
@@ -0,0 +1,80 @@
+@description('Specifies the name of the Azure Machine Learning workspace where sweep job will be deployed')
```
Comment (shivamMg) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep:1]: I think these warnings from the Pipeline can be fixed:

```
D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job\modules\codeversion.bicep(44,9) : Warning use-stable-resource-identifiers: Resource identifiers should be reproducible outside of their initial deployment context. Resource deploymentScript's 'name' identifier is potentially nondeterministic due to its use of the 'utcNow' function (deploymentScript.name -> utcValue (default value) -> utcNow()). [https://aka.ms/bicep/linter/use-stable-resource-identifiers]
D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job\modules\codeversion.bicep(72,9) : Warning use-stable-resource-identifiers: Resource identifiers should be reproducible outside of their initial deployment context. Resource codeVersionResource's 'name' identifier is potentially nondeterministic due to its use of the 'utcNow' function (codeVersionResource.name -> codeId (default value) -> utcValue (default value) -> utcNow()). [https://aka.ms/bicep/linter/use-stable-resource-identifiers]
D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job\modules\codeversion.bicep(74,50) : Warning no-hardcoded-env-urls: Environment URLs should not be hardcoded. Use the environment() function to ensure compatibility across clouds. Found this disallowed host: "core.windows.net" [https://aka.ms/bicep/linter/no-hardcoded-env-urls]
```

What do you think?

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    scriptContent: 'echo "$CONTENT" > ${filename} && az storage blob upload -f ${filename} -c ${containerName} -n ${filename}'
+  }
+}
+
+// creating codeVersion resource using uploaded hyperdrive script blob url
+resource codeVersionResource 'Microsoft.MachineLearningServices/workspaces/codes/versions@2022-05-01' = {
+  dependsOn: [ deploymentScript ]
+  name: '${workspaceName}/${codeId}/${codeVersion}'
+  properties: {
+    codeUri: 'https://${storageAccountName}.blob.core.windows.net/${containerName}/'
```
Comment (shivamMg) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep:74]: Use the uri function? I think you should be able to get rid of this warning in Pipelines:

```
##[error]        Function 'format' found within 'codeUri'
```

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep [hunk 4]
```diff
@@ -0,0 +1,119 @@
+// Sweep Job Resource using ARM Template
```
Comment (shivamMg) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:1]: Do you know anything about this error from Pipelines:

```
##[error]D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job\main.bicep(1,1) : Warning linter-internal-error: Analyzer 'core/artifacts-parameters' encountered an unexpected exception. Attempted to access a missing method.
```

**Review by shivamMg — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep (hunk 4 above)
Comment (Anubha98) [reply quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:1]: Not sure, tried removing comment as well, also making sure usage defined here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter-rule-artifacts-parameters warning still persists.

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/script/main.py (hunk 1 above)
Comment (Anubha98) [reply quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/script/main.py:1]: Yes, Done.

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep (hunk 2 above)
Comment (Anubha98) [reply quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep:1]: Done.

**Review by shivamMg — APPROVED** [review]
Would be great to understand and fix these two from the Azure Pipelines:

```
D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job\main.bicep(1,1) : Warning linter-internal-error: Analyzer 'core/artifacts-parameters' encountered an unexpected exception. Attempted to access a missing method.

##[error]Test-FileList : Could not extract inner templates for 'D:\a\1\s\quickstarts\microsoft.machinelearningservices\machine-learning-job-create-sweep-job'.
```

Rest lgtm.

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      name: containerName
+      properties: {
+        publicAccess: 'Container'
+      }
+    }
+  }
+}
+
+// unique value based on container id
+var uniqueIdentifier = uniqueString(storage::blobService::container.id)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/modules/codeversion.bicep:39]: move all var declarations above the resources

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+@description('Specifies the curated environment to run sweep job.')
+param environmentName string = 'AzureML-lightgbm-3.2-ubuntu18.04-py37-cpu'
+
+// fetching existing curated environment resource object
+resource environmemt 'Microsoft.MachineLearningServices/workspaces/environments@2022-05-01' existing = {
+  name: '${workspaceName}/${environmentName}'
+}
+
+// fetching exisitng compute/ environment resource ids
+var computeId = resourceId('Microsoft.MachineLearningServices/workspaces/computes', workspaceName, computeName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:72]: move all var declarations above resources

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+var codeId = codeVersion.outputs.codeId
+
+// creating sweep job resource
+resource sweepjobResource 'Microsoft.MachineLearningServices/workspaces/jobs@2022-06-01-preview' = {
+  name: '${workspaceName}/${jobName}'
+  properties: {
+    description: 'Sweep Job Resource from ARM Template'
+    properties: {}
+    tags: {
+      referenceNotebook: 'https://github.com/Azure/azureml-examples/blob/main/sdk/jobs/single-step/lightgbm/iris/lightgbm-iris-sweep.ipynb'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:95]: what happens if this file is not found?

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep (hunk 4 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:1]: I think this is a bicep bug, nothing related to the sample...

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep (hunk 7 above)
Comment (shivamMg) [reply quickstarts/microsoft.machinelearningservices/machine-learning-job-create-sweep-job/main.bicep:95]: It's just a tag that links to a notebook that creates a similar job as this arm template. It shows up at the AzureML Studio UI like this:

image

The link being invalid (e.g. 404) will not fail the job run.

**Pull request merged**